src/renderer/coremods/contextMenu/index.tsx

```diff
@@ -84,57 +84,74 @@ type ContextMenuData = ContextMenuProps["ContextMenu"] & {
-  if (menu.plugged) return;
-
-  // We delay getting the items until now, as importing at the start of the file causes Discord to hang
-  // Using `await import(...)` is undesirable because the new items will only appear once the menu is interacted with
-  const { MenuGroup } = getByProps<Record<string, React.ComponentType>>([
-    "Menu",
-    "MenuItem",
-    "MenuGroup",
-  ])!;
-
-  if (!MenuGroup) return;
+  // Or MenuGroup Component is not available
+  if (!menuItems[navId] || !MenuGroup) return <ContextMenu {...menu} plugged={true} />;
 
   // The data as passed as Arguments from the calling function, so we just grab what we want from it
   const data = menu.data[0];
 
-  const repluggedGroup = <MenuGroup />;
-  repluggedGroup.props.id = "replugged";
-  repluggedGroup.props.children = [];
+  //Add group only if it doesn't exist
+  if (!menu.children.some((child) => child?.props?.id === "replugged")) {
+    const repluggedGroup = <MenuGroup />;
+    repluggedGroup.props.id = "replugged";
+    repluggedGroup.props.children = [];
+
+    // Add in the new menu items right above the DevMode Copy ID
+    // If the user doesn't have DevMode enabled, the new items will be at the bottom
+    menu.children.splice(-1, 0, repluggedGroup);
+  }
+
+  //get sections from where to clean items
+  const usedSectionIds = menuItems[navId]
+    .map((item) => item.sectionId)
+    .filter((item, index, array) => array.indexOf(item) === index);
 
-  // Add in the new menu items right above the DevMode Copy ID
-  // If the user doesn't have DevMode enabled, the new items will be at the bottom
-  menu.children.splice(-1, 0, repluggedGroup);
+  //cleaning old items before adding new ones
+  usedSectionIds.forEach((sectionId) => {
+    try {
+      if (menu.children.at(sectionId)?.props?.children) {
+        menu.children.at(sectionId)!.props.children = Array.isArray(
+          menu.children.at(sectionId)?.props.children,
+        )
+          ? menu.children
+              .at(sectionId)
+              ?.props.children.filter((child: React.ReactElement) => child.props.plugged)
+          : [menu.children.at(sectionId)?.props.children];
+      }
+    } catch (err) {
+      logger.error("Error while removing old menu items", err, menu.children.at(sectionId));
+    }
+  });
 
+  //adding new items
   menuItems[navId].forEach((item) => {
     try {
-      const res = makeItem(item.getItem(data, menu)) as ContextItem & { props: { id?: string } };
+      const itemRet = makeItem(item.getItem(data, menu)) as ContextItem & {
+        props: { replug?: boolean };
+      };
+
+      // removed the code to add unique id every time because it made experience bad while hovering over item and it getting updated
+      // should be fine because we actively keep removing old items ig
 
-      if (res?.props) {
-        // add in unique ids
-        res.props.id = `${res.props.id || "repluggedItem"}-${Math.random()
```

Comment:
Can you add the unique id back. In what way "it made experience bad while hovering over item and it getting updated". I'm not having any "bad experience" with it.

Reply:
![haha-ye.gif](https://github.com/replugged-org/replugged/assets/73281112/08d88259-ffa4-4768-8fba-b4666208fd67)

This is how ig. you need an item that gets updated and if you add unique ID everytime then it will be different from what hover had in its state and this will happen

Reply:
I have tested and is not changing id.

Reply:
> I have tested and is not changing id.

you updating the menu?

Reply:
Unfortunately we need to have the unique id, there might be some plugins that use the same id...

Reply:
> Unfortunately we need to have the unique id, there might be some plugins that use the same id...

plugin devs should address that not the client mod, we should value user experience more.
idk will have to think of some way to store it or anything then to keep old id's maybe a map on the menu props?

Reply:
This will result in a breaking change if you remove the unique id, there are already some plugins that are adding items without an id because we were handling that with the API. Yes, we need to find a better solution to save the id, take a look at how it has been done with message popover API.

Reply:
even there it's not a unique ID always. either they provide a id or random ID. we can do that here too I mean but unique always is just meh kinda but we can store them on the menu as it stays constant just the children gets changed and the plugged value on it